Create Conversion-Class Temperatura

StelaraStina opened the pull request:
And this is in Main:

package com.company;

public class Main {

    public static void main(String[] args) {
        Temperatura temperatura = new Temperatura ();
        temperatura.temperatura(0,70);
        System.out.println("Temperatura zilei de azi este de: " + temperatura.temperatura + " grade celsius.");
    }
}

Could you write notes as comments with // ? Just tell me where my thinking is wrong, i would like to build the code myself. I just need a walkthrough.

StelaraStina committed:
Create Conversion-Class Temperatura

MirelaI commented on Conversion-Class Temperatura:
```diff
@@ -0,0 +1,18 @@
+        return temperatura;
+    }
+
+    public void temperatura (double c, double f) {
```
Rename to `conversion`, it conflicts with the attribute `temperatura`

MirelaI replied:
The naming of the function should also contain what kind of conversion it is, for example `conversionCelsiusToFahrenheit`, or the other way around, depending on your function purpose

MirelaI commented on Conversion-Class Temperatura:
```diff
@@ -0,0 +1,18 @@
+    public void temperatura (double c, double f) {
+        double celsius = 0;
+        double fahrenheit = 70;
+        double temperatura = 5/9*(f-32);
```
Where are you using the `c` parameter? If you want to override the temperatura attribute in this final line you can do it with the setter defined above `setTemperatura`

MirelaI replied:
If you want to convert the `temperatura`, you can get the full use of your getter and setter defined above. No need of any arguments to be passed. If can be something like this: 
`this.setTemperatura(5/9*(this.temperatura - 32))`

MirelaI commented on Conversion-Class Temperatura:
```diff
@@ -0,0 +1,18 @@
+
+    public void temperatura (double c, double f) {
+        double celsius = 0;
+        double fahrenheit = 70;
```
`celsius` and `fahrenheit` are not used anywhere in this function, do you think you can remove them, or you need to use them somewhere else?

StelaraStina committed:
Update Conversion-Class Temperatura

package com.company;

public class Main {

    public static void main(String[] args) {
      
Temperatura conversionfromFtoC = new Temperatura ();
        conversionfromFtoC.setTemperatura(50);
        System.out.println("Temperatura zilei de azi este de: " + conversionfromFtoC + " grade celsius.");
    }
}

And i get the result 

Temperatura zilei de azi este de: com.company.Temperatura@1540e19d grade celsius.

Process finished with exit code 0

MirelaI commented:
In your Main app, you need to set first the attribute temperatura

temperatura.temperatura(70), now your object temperatura has an attribute temperatura which is set to 70.

Now you want to convert it and in order to do that you need to call on the object the conversion function.

The conversion function `conversionfromFtoC` returns the converted value.

If you want to display it in your System.out.println you do something like: 
`System.out.println("Temperatura zilei de azi este de: " + temperatura.conversionfromFtoC + " grade celsius.");`

Note here that calling the object the same as the attribute might get confusing when debugging. Try to rename the object into something like 
`  Temperatura temperatureObj = new Temperatura ();`

MirelaI commented on Conversion-Class Temperatura:
```diff
@@ -0,0 +1,16 @@
+    }
+
+    public void conversionfromFtoC () {
+        double setTemperatura = 5/9*(this.temperatura-32);
```
Variable names should not include verbs like `set` since that implies there is an action happening.
so rename your variable into `convertedTemperature`

MirelaI commented on Conversion-Class Temperatura:
```diff
@@ -0,0 +1,16 @@
+        return temperatura;
+    }
+
+    public void conversionfromFtoC () {
```
return a double here, since the conversion should return the value of the conversion

MirelaI commented on Conversion-Class Temperatura:
```diff
@@ -0,0 +1,16 @@
+
+    public void conversionfromFtoC () {
+        double setTemperatura = 5/9*(this.temperatura-32);
+    }
```
add a return statement inside the function
`return convertedTemperature`

StelaraStina replied:
This is my main now 

package com.company;

public class Main {

    public static void main(String[] args) {

        Temperatura conversionfromFtoC = new Temperatura ();
        conversionfromFtoC.setTemperatura(50);
        System.out.println("Temperatura zilei de azi este de: " + conversionfromFtoC + " grade celsius.");
    }
}

MirelaI replied:
`
public static void main(String[] args) {

    Temperatura tempObj = new Temperatura ();
    conversionfromFtoC.setTemperatura(50);
    System.out.println("Temperatura zilei de azi este de: " + tempObj.conversionfromFtoC + " grade celsius.");
}
`

StelaraStina committed:
Update Conversion-Class Temperatura

StelaraStina committed:
Update Conversion-Class Temperatura

MirelaI commented on Conversion-Class Temperatura:
```diff
@@ -0,0 +1,14 @@
+public class Temperatura {
+
+    public double temperatura;
+    public void setTemperatura(double temperaturaZilei) {
```
you also need to put back again the getter for the `temperatura`: `getTemperatura`

MirelaI commented on Conversion-Class Temperatura:
```diff
@@ -0,0 +1,14 @@
+    }
+
+    public double conversionfromFtoC () {
+        double convertedTemperatura = 5/9*(this.temperatura-32);
```
here this.temperatura to be replaced with this.getTemperatura

StelaraStina committed:
Update Conversion-Class Temperatura

StelaraStina commented:
This is the main:

package com.company;

public class Main {

    public static void main(String[] args) {

      Temperatura conversionfromFtoC = new Temperatura ();
        conversionfromFtoC.setTemperatura(20);
        System.out.println("Temperatura zilei de azi este de: " + conversionfromFtoC + " grade celsius.");
    }
}

MirelaI commented:
Just one minor change on the Main class: 
conversionfromFtoC must be called on the Temperatura object 
conversionfromFtoC. conversionfromFtoC

`
public class Main {

public static void main(String[] args) {

  Temperatura tempObj = new Temperatura ();
    tempObj.setTemperatura(20);
    System.out.println("Temperatura zilei de azi este de: " + tempObj.conversionfromFtoC + " grade celsius.");
}

`
Try it as above and see what result are you getting

StelaraStina commented on Conversion-Class Temperatura:
```diff
@@ -0,0 +1,18 @@
+        double celsius = 0;
+        double fahrenheit = 70;
+        double temperatura = 5/9*(f-32);
+    }
```
![temperature warning](https://user-images.githubusercontent.com/29537741/27330148-e01b1c16-55c0-11e7-813e-b8b8372c6611.jpg)
![temperature](https://user-images.githubusercontent.com/29537741/27330149-e02721f0-55c0-11e7-886f-e8f13b1214d2.JPG)
![main](https://user-images.githubusercontent.com/29537741/27330147-e00b7964-55c0-11e7-8f2a-3320a1e33406.JPG)

StelaraStina replied:
I am going crazy @MirelaI :(( #2
